src/main/java/org/web3j/tx/ManagedTransaction.java

```diff
@@ -75,8 +81,13 @@ protected TransactionReceipt processResponse(EthSendTransaction transactionRespo
         }
 
         String transactionHash = transactionResponse.getTransactionHash();
-
-        return waitForTransactionReceipt(transactionHash);
+        if (attempts > 0) {
```

Comment:
What sort of transaction volumes per minute would you like this to support? Given the current limitations in Ethereum on transaction throughput, I think the polling is a relatively unobtrusive operation.